src/Molecules/UserSelect/UserSelect.test.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState } from 'react'
+import { render, fireEvent } from '@testing-library/react'
+import user from '@testing-library/user-event'
+import '@testing-library/jest-dom/extend-expect'
+
+import UserSelect, { User } from './index'
+
+// required until jsdom>=16 (https://github.com/mui-org/material-ui/issues/15726#issuecomment-493124813)
+global.document.createRange = () => ({
+  setStart:                () => {},
+  setEnd:                  () => {},
+  commonAncestorContainer: {
+    nodeName:      'BODY',
+    ownerDocument: document,
+  },
+})
+
+const StateHolder = ({ children = () => {}, state = [] }) => {
+  const [values, onChange] = useState(state)
+  const [errors, setErrors] = useState([])
+  const onError = data => setErrors(errors.concat(data))
+
+  return <>
+    <div data-testid="values">
+      {values.map(value => <div key={value} data-testid={`value-${value}`} />)}
+    </div>
+    <div data-testid="errors">
+      {errors.map(({ error }) => <div key={error} data-testid={`error-${error}`} />)}
+    </div>
+    {children({ values, onChange, onError })}
+  </>
+}
+
+it('should show label when rendered', () => {
+  const { getByText } = render(
+    <UserSelect label="Hello world" />,
+  )
+
+  expect(getByText('Hello world')).toBeVisible()
+})
+
+it.each([
+  '//material-ui.com/static/images/avatar/1.jpg',
+  'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7',
+])('should display avatar when provided', (avatar, id = 1) => {
+  const { getByAltText } = render(
+    <UserSelect values={[id]}>
+      <User value={id} avatar={avatar} />
+    </UserSelect>,
+  )
+
+  expect(getByAltText(avatar)).toBeVisible()
+})
+
+it.each([
+  'success',
+  'warning',
+])('should display correct color for a given status when provided', (status, id = 1) => {
+  const { getByText } = render(
+    <UserSelect values={[id]}>
+      <User value={id} status={status}>{status}</User>
+    </UserSelect>,
+  )
+
+  expect(getByText(status).parentNode).toHaveAttribute('class', expect.stringContaining(`-${status}-`))
+})
+
+it('should add item in the list', async () => {
+  const { getByRole, getByTestId, findByText, findByTestId } = render(
+    <StateHolder>
+      {({ ...props }) => (
+        <UserSelect {...props}>
+          <User value={1}>user</User>
+        </UserSelect>
+      )}
+    </StateHolder>,
+  )
+
+  user.click(getByRole('combobox'))
+  user.click(await findByText('user'))
+
+  expect(getByTestId('values')).toContainElement(await findByTestId('value-1'))
```

Comment:
Can't we directly check inside the component itself for the added value instead of testing the `StateHolder` ?

Reply:
This is supposed to be the corrected way to test state in a UI way according to @mesteche